addons/SimpleTODO/TODOItem.gd

```diff
@@ -4,18 +4,184 @@ extends HBoxContainer
+			initial_item_index = get_index()
+			
+			get_parent().remove_child(self)
+			
+			parent_column.update_counter()
+			
+			rect_min_size = rect_size
+			
+			# Set the size vertical flags to none so that it doesn't stretch
+			# when being dragged
+			size_flags_vertical = 0
+			
+			main.add_child(self)
+			
+			# Set dragging to true to tell _process to now handle dragging
+			is_dragging = true
+
+			item_placement_holder.get_parent().remove_child(item_placement_holder)
+			parent_column.item_container.add_child(item_placement_holder)
+			item_placement_holder.get_parent().move_child(item_placement_holder, initial_item_index)
+			
+			item_placement_holder.visible = true
+			item_placement_holder.rect_min_size = rect_min_size
+			item_placement_holder.rect_size = rect_size
+
+# Handles left click being released
+func _input(event):
+	if event is InputEventMouseButton:
+		if event.button_index == 1 and not event.pressed and is_dragging:
+			# Set dragging to false to tell _process that we are no
+			# longer dragging this todo item
+			is_dragging = false
+			
+			get_parent().remove_child(self)
+			
+			size_flags_vertical = SIZE_FILL
+
+			rect_min_size = Vector2.ZERO
+			
+			if item_placement_holder:
+				item_placement_holder.visible = false
+				item_placement_holder.get_parent().remove_child(item_placement_holder)
+				main.add_child(item_placement_holder)
+
+			move_item(current_drag_item_index)
+
+			current_drag_item_index = 0
+			initial_item_index = 0
+
+# Used for undo redo, if we can't remove this item then don't error out
+func remove_self_safe():
+	var parent = get_parent()
+	if parent:
+		parent.remove_child(self)
+
+func move_item(index):
+	undo_redo.create_action("Move Item")
+
+	var current_index = initial_item_index
+	
+	undo_redo.add_do_method(self, "remove_self_safe")
+	undo_redo.add_do_method(next_parent_column.item_container, "add_child", self)
+	undo_redo.add_do_method(next_parent_column.item_container, "move_child", self, index)
+	undo_redo.add_do_method(next_parent_column, "request_save")
+	undo_redo.add_do_property(self, "parent_column", next_parent_column)
+	undo_redo.add_do_property(self, "next_parent_column", parent_column)
+	
+	undo_redo.add_undo_method(self, "remove_self_safe")
+	undo_redo.add_undo_method(parent_column.item_container, "add_child", self)
+	undo_redo.add_undo_method(parent_column.item_container, "move_child", self, current_index)
+	undo_redo.add_undo_method(parent_column, "request_save")
+	undo_redo.add_undo_property(self, "parent_column", parent_column)
+
+	undo_redo.commit_action()
+
+func delete_item():
+	undo_redo.create_action("Delete Item")
+	undo_redo.add_do_method(parent_column.item_container, "remove_child", self)
+	undo_redo.add_do_method(parent_column, "request_save")
+	undo_redo.add_undo_method(parent_column.item_container, "add_child", self)
+	undo_redo.add_undo_method(parent_column, "request_save")
+	undo_redo.commit_action()
```

Comment:
This method could've stayed in the column, you could just call it from here.

Reply:
Yes I could of kept it in the column though it would then be a little bit harder to keep track of where the item is add, especially when dealing with the undo redo system. That's why I moved it to the item script instead.